### ISSUE TITLE
NKT_mean output Nan, when the number of training sample is increased

### ISSUE THREAD

**Issue by zhangbububu** [google/neural-tangents#198]
hi, i meet a confuse problem

```
init_fn, apply_fn, kernel_fn = stax.serial(
    stax.Dense(512, W_std=1.5, b_std=0.05), stax.Relu(do_stabilize=True),
    stax.Dense(512, W_std=1.5, b_std=0.05), stax.Relu(do_stabilize=True),
    stax.Dense(1, W_std=1.5, b_std=0.05)
)

s = 10
l = jnp.pi * -s
r = jnp.pi * s 
N_tr = 100
N_te = 5
train_xs = jnp.linspace(l, r , N_tr).reshape(-1, 1).astype(jnp.float64)
train_ys = jnp.sin(train_xs) + jnp.sin(2*train_xs).astype(jnp.float64)
test_xs = jnp.linspace(l, r, N_te).reshape(-1, 1).astype(jnp.float64)

predict_fn = nt.predict.gradient_descent_mse_ensemble(kernel_fn, train_xs,
                                                      train_ys, diag_reg=1e-4)
nkt_mean, nkt_covariance = predict_fn(x_test=test_xs, get='ntk',
                                        compute_cov=True)
print(nkt_mean)


```

if i increate the number of training samples (N_tr), i will get a all NaN `nkt_mean`

![image](https://github.com/google/neural-tangents/assets/41133441/46acabcc-90c5-4385-b25e-1a2af2a609db)
![image](https://github.com/google/neural-tangents/assets/41133441/4d1405ea-8d66-4369-b321-67a253c2766c)

**Comment by romanngg** [comment]
I think the reason is that this 1D function is hard to fit with a Relu kernel, but sampling only 15 points makes it a simpler training objective, so it fits it with a lower diagonal regularizer. You can avoid NaNs by increasing `diag_reg` which I did below, but as you can see it's a poor fit in any case. (NTK prediction is orange with 1000 test points sampled).

1000 training points, `diag_reg=1e-2`:
<img width="380" alt="download3" src="https://github.com/google/neural-tangents/assets/44512421/02a67a65-bb66-4f5d-b9a6-3819816b2b6c">
100 training points, `diag_reg=1e-3`:
<img width="380" alt="download" src="https://github.com/google/neural-tangents/assets/44512421/42684a10-451a-48f2-accf-d3a7a2f717d7">
15 training points, `diag_reg=1e-4`:
<img width="371" alt="download2" src="https://github.com/google/neural-tangents/assets/44512421/e8a5692e-44b5-4399-b9c9-daab854fbfa3">

**Comment by zhangbububu** [comment]
@romanngg 

Thank you very much for your careful answer.

I am currently doing similar experiments. Can you tell me some ways to make NKT fit better ?

**Comment by romanngg** [comment]
I guess for this particular example, knowing your training targets, a periodic nonlinearity would fit better (`stax.Sin(), diag_reg=1e-4`):

<img width="380" alt="sin" src="https://github.com/google/neural-tangents/assets/44512421/e8ff4b0d-d39f-48cc-abc7-0317d64a4e9a">

Otherwise trying different architectures and plotting predictions or draws from the prior would be good to gain intuition for what works best. Note that for time series data of shape `[batch_size, time_duration, n_features]`, I imagine you may want to use 1D-convolution `stax.Conv`/`stax.ConvLocal` over the `time_duration` axis, to incorporate time locality into your model.
